tsdb/head.go

```diff
@@ -1262,6 +1262,34 @@ func (h *Head) IsQuerierCollidingWithTruncation(querierMint, querierMaxt int64)
 	return false, false, 0
 }
 
+func (h *Head) checkDeleted(id chunks.HeadSeriesRef) (int, bool) {
```

Comment:
For reviewers: this was refactoring done for a previous implementation, but I kept it since it makes the code a bit more readable, in my opinion.